lib/change_history_modifier.rb

```diff
@@ -0,0 +1,72 @@
+class ChangeHistoryModifier
+  def initialize(content_id, locale, user_email)
+    @content_id = content_id
+    @locale = locale || "en"
+    @user = User.find_by!(email: user_email) if user_email
+  end
+
+  def delete(id)
+    update_edition do |edition, updater|
+      entry = edition.change_history.find { |item| item["id"] == id }
+      raise "No change history entry with id #{id}" unless entry
+
+      updater.assign(change_history: edition.change_history - [entry])
+    end
+  end
+
+  def edit(id, note)
+    update_edition do |edition, updater|
+      original_entry = edition.change_history.find { |item| item["id"] == id }
+      raise "No change history entry with id #{id}" unless original_entry
+
+      updated_entry = original_entry.dup
+      updated_entry["note"] = note
+
+      updater.assign(
+        change_history: sort_change_history(
+          edition.change_history - [original_entry] + [updated_entry]
```

Comment:
Please let me know if there is a better way to do this as this looks a bit hacky!  But, it is the only way I've managed to make this work.

Reply:
See comment below on this line...